### DISCUSSION TITLE
Create a `.flyteignore` to ignore additional files whilst fast-registering

### DISCUSSION THREAD

**Discussion by bstadlbauer** [flyteorg/flyte#4734] [RFC Incubator]
When a user fast registers, all files which are not in `.gitignore` or `.dockerignore` will be packaged up and sent to Flyte ([docs](https://docs.flyte.org/en/latest/flytesnacks/getting_started/package_register.html#fast-registration)). This can lead to unexpected large tarballs and thus long registrations times when users have large files (which might not be under version control) in their project directory.
While users can all those files to `.dockerignore` or to `.gitignore`, this will always result in the expected side-effects of those tools.

**Comment by fg91** [comment]
+1 We certainly were in situations as well where people added certain files/dirs to their .gitignore/.dockerignore for Flyte's fast registration but then couldn't actually commit those changes to .gitignore/.dockerignore.

While I think it's a good idea to implement this, I don't think it requires an RFC.

**Comment by davidmirror-ops** [comment]
Thanks for bringin this up @bstadlbauer 
As discussed in today's contributors meetup, this is a good idea but doesn't need an RFC.
Please transfer to an Issue and continue there.

**Comment by fg91** [comment]
Closing this discussion since this has been implemented, see [here](https://github.com/flyteorg/flytekit/blob/65bf9e7896285d290ab1f1698c98cc75e623d11d/flytekit/tools/ignore.py#L91).
